control/sysctl.go

```diff
@@ -0,0 +1,90 @@
+package control
+
+import (
+	"os"
+	"strings"
+	"sync"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+)
+
+const SysctlPrefixPath = "/proc/sys/"
+
+var sysctl *SysctlManager
+
+type SysctlManager struct {
+	mux          sync.Mutex
+	watcher      *fsnotify.Watcher
+	expectations map[string]string
+}
+
+func init() {
+	var err error
+	if sysctl, err = NewSysctlManager(); err != nil {
+		logrus.Fatalf("failed to create sysctl manager: %v", err)
+	}
+}
+
+func NewSysctlManager() (*SysctlManager, error) {
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	manager := &SysctlManager{
+		mux:          sync.Mutex{},
+		watcher:      watcher,
+		expectations: map[string]string{},
+	}
+	go manager.StartWatch()
+	return manager, nil
+}
+
+func (s *SysctlManager) StartWatch() {
+	for {
+		select {
+		case event, ok := <-s.watcher.Events:
+			if !ok {
+				return
+			}
+			if event.Has(fsnotify.Write) {
+				logrus.Tracef("sysctl write event: %+v", event)
```

Comment:
Don't use global logrus object.

Reply:
So I have to delete a log at udp.go:sendPkt() where I don't find a trivial way to get a logrus.Logger. Hope that doesn't matter in the aspect of debugging.

Reply:
Pass in log as the first param is acceptable if necessary.